hypemaths/models/matrix.py

```diff
@@ -30,6 +33,80 @@ def dims(self) -> tuple:
     def __repr__(self) -> str:
         return f"{self.__class__.__name__}({self.matrix})"
 
+    def multiply_by_number(self, number: t.Union[int, float]) -> 'Matrix':
+        '''
+        Multiplies the current matrix with a number.
+        '''
+        if type(number) is not int:
+            raise TypeError(f"Expected `int` but got {type(number).__name__} ({number})")
```

Comment:
```suggestion
        if not isinstance(number, int):
            raise TypeError(f"Expected `int` but got {type(number)}")
```
This is an easier way to handle types :)